### PR TITLE
Fix margins in UIViewBox

### LINKED ISSUE #2003: Box margins entirely not functioning on iOS
Looks like margins entirely not functioning on iOS. Going to file that for follow-up.

_Originally posted by @JakeWharton in https://github.com/cashapp/redwood/issues/1838#issuecomment-1973872950_

### PR DESCRIPTION
**Pull request by swankjesse**
Closes: https://github.com/cashapp/redwood/issues/2003

---

- [x] `CHANGELOG.md`'s "Unreleased" section has been updated, if applicable.

### CODE REVIEW

File: redwood-layout-composeui/src/test/snapshots/images/app.cash.redwood.layout.composeui_ComposeUiBoxTest_testMarginsAndAlignment[LTR].png
Comment (swankjesse) [inline redwood-layout-composeui/src/test/snapshots/images/app.cash.redwood.layout.composeui_ComposeUiBoxTest_testMarginsAndAlignment[LTR].png:1]: This is broken! We don’t implement margins in ComposeUI

File: redwood-layout-uiview/RedwoodLayoutUIViewTests/__Snapshots__/UIViewBoxTestHost/testChildrenModifierChanges.Margin.png
Comment (swankjesse) [inline redwood-layout-uiview/RedwoodLayoutUIViewTests/__Snapshots__/UIViewBoxTestHost/testChildrenModifierChanges.Margin.png:1]: Now consistent with Android:
https://github.com/cashapp/redwood/blob/trunk/redwood-layout-view/src/test/snapshots/images/app.cash.redwood.layout.view_ViewBoxTest_testChildrenModifierChanges%5BLTR%5D_margin.png

File: redwood-layout-uiview/RedwoodLayoutUIViewTests/__Snapshots__/UIViewBoxTestHost/testMargins.1.png
Comment (swankjesse) [inline redwood-layout-uiview/RedwoodLayoutUIViewTests/__Snapshots__/UIViewBoxTestHost/testMargins.1.png:1]: Now consistent with Android:
https://github.com/cashapp/redwood/blob/trunk/redwood-layout-view/src/test/snapshots/images/app.cash.redwood.layout.view_ViewBoxTest_testMargins%5BLTR%5D.png

File: redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewBox.kt [hunk 1]
```diff
@@ -191,103 +200,162 @@ internal class UIViewBox :
...
+internal fun maxEachDimension(a: CValue<CGSize>, b: CValue<CGSize>): CValue<CGSize> {
+  return a@a.useContents {
+    b@b.useContents {
+      CGSizeMake(
+        maxOf(a@width, b@width),
+        maxOf(a@height, b@height),
+      )
+    }
+  }
+}
```
Comment (swankjesse) [inline redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/UIViewBox.kt:353]: This was the least worst of a few different ways of writing this function. Open to feedback!

File: redwood-layout-view/src/test/snapshots/images/app.cash.redwood.layout.view_ViewBoxTest_testMarginsAndAlignment[LTR].png
Comment (swankjesse) [inline redwood-layout-view/src/test/snapshots/images/app.cash.redwood.layout.view_ViewBoxTest_testMarginsAndAlignment[LTR].png:1]: Note that 'center both' is deliberately not perfectly centered; the widget has asymmetric margins

File: redwood-layout-composeui/src/test/snapshots/images/app.cash.redwood.layout.composeui_ComposeUiBoxTest_testMarginsAndAlignment[LTR].png
Comment (JakeWharton) [reply redwood-layout-composeui/src/test/snapshots/images/app.cash.redwood.layout.composeui_ComposeUiBoxTest_testMarginsAndAlignment[LTR].png:1]: https://github.com/cashapp/redwood/issues/2006

**Review by JakeWharton — APPROVED** [review]

**Pull request merged**